Jr/feature segement modals

jeffrummel opened the pull request.

jeffrummel committed:
removes extra "→"

jeffrummel committed:
adds and refines necessary files

jeffrummel committed:
Merge branch 'master' into jr/feature-segement-modals

jeffrummel committed:
design refinements -- rwd

jeffrummel committed:
updates image usage

jeffrummel committed:
completes merge

earthboundkid commented on data/segments.json:
```diff
@@ -0,0 +1,39 @@
+{
+  "newsletterUserActive": "true",
```
Change these to `true` (no quotes) and change the templates to `if $data.newsletterUserActive` etc.

earthboundkid commented on layouts/partials/tw/modal-takeover.html:
```diff
@@ -43,27 +49,41 @@
+          fill-yellow-8
+        {{ end }} {{ with eq $modalType "donate" }}
+          fill-red-8
+        {{ end }} fill-red-8
```
fill-red-8 is repeated here and should be removed.

earthboundkid commented on layouts/partials/tw/modal-sticky-anchored.html:
```diff
@@ -0,0 +1,119 @@
+{{ $magicPixel := "data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" | safeURL }}
```
This is unused.

jeffrummel committed:
removes magicpixel -- https://github.com/spotlightpa/poor-richard/pull/645/files/5d94b85f4aa611e591cc56922d880a292167b822#r1600644882

jeffrummel committed:
removes extra fill-red-8 -- https://github.com/spotlightpa/poor-richard/pull/645/files/5d94b85f4aa611e591cc56922d880a292167b822#r1600643747

jeffrummel committed:
sets segment active to boolean -- https://github.com/spotlightpa/poor-richard/pull/645#discussion_r1600343124

jeffrummel committed:
Merge branch 'master' into jr/feature-segement-modals

earthboundkid commented on layouts/partials/tw/modal-sticky-window.html:
```diff
@@ -0,0 +1,128 @@
+{{ $magicPixel := "data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" | safeURL }}
```
Here's another one.

earthboundkid commented on layouts/partials/tw/modal-ribbon.html:
```diff
@@ -1,6 +1,15 @@
+{{ $magicPixel := "data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" | safeURL }}
```
And one more.

earthboundkid commented on layouts/partials/tw/modal-takeover.html:
```diff
@@ -1,32 +1,38 @@
-{{ $dek := page.Param "takeover-dek" }}
-{{ $link := page.Param "takeover-link" }}
+{{ $imagename := .filename }}
+{{ $magicPixel := "data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" | safeURL }}
```
This one too.

jeffrummel committed:
removes magicpixel:

- https://github.com/spotlightpa/poor-richard/pull/645#discussion_r1603498295
- https://github.com/spotlightpa/poor-richard/pull/645#discussion_r1603495012
- https://github.com/spotlightpa/poor-richard/pull/645#discussion_r1603494774

jeffrummel committed:
Merge branch 'jr/feature-segement-modals' of github.com:spotlightpa/poor-richard into jr/feature-segement-modals

jeffrummel committed:
Merge branch 'master' into jr/feature-segement-modals

earthboundkid commented on layouts/partials/tw/modal-sticky-anchored.html:
```diff
@@ -0,0 +1,118 @@
+{{ $gaForm :=  "newsletters:papost-full" }}
```
gaForm and gaCategory should match the .hiddenField. Maybe just default to PA Post but take .gaForm and .gaCategory if it exists.

jeffrummel committed:
sets $gaForm with default

jeffrummel committed:
Merge branch 'jr/feature-segement-modals' of github.com:spotlightpa/poor-richard into jr/feature-segement-modals
